### PR TITLE
Add a rule that forbids styling types directly

### PR DESCRIPTION
**Pull request by roookeee**
As discussed in #68 I have implemented a rule that forbids styling types directly. I changed the wording from tag to type for consistencies sake in regards to the UnknownTypeSelectorRule.

### CODE REVIEW

**Comment by roookeee** [comment]
The travis error is not related to issues with my PR as the console output shows that the maven build was successful.
I really didn't know what to write on the rules description page. Tell me if I can improve it somewhere.

File: css-checks/src/main/resources/org/sonar/css/checks/l10n/common/template/type-selector.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+.mybox div {
+  color: red;
+}
+</pre>
+
+<p>
+    You can exclude certain types via the configuration of this rule as e.g. the anchor type is a valid exception from this rule as it should always denote a link of some sort.
+</p>
```
Comment (racodond) [inline css-checks/src/main/resources/org/sonar/css/checks/l10n/common/template/type-selector.html]: @naze: Could you please add an empty line at the end of the file?

**Review by racodond — COMMENTED** [review]
@naze: Pretty good start! Thanks!

Did you make this rule work in real life? I mean unit testing apart.
For this rule to be available, you need to declare it at:
* https://github.com/racodond/sonar-css-plugin/blob/master/css-checks/src/main/java/org/sonar/css/checks/CheckList.java#L114
* https://github.com/racodond/sonar-css-plugin/blob/master/css-checks/src/main/java/org/sonar/css/checks/CheckList.java#L197

Could you also pull request on https://github.com/racodond/sonar-css-plugin/tree/naze instead? So that, I can fine tune the rule on my side: by adding integration tests, etc.

Thanks!

David

**Comment by roookeee** [comment]
> Did you make this rule work in real life? I mean unit testing apart.

What do mean with this?

> Could you also pull request on https://github.com/racodond/sonar-css-plugin/tree/naze instead? 

As I can not change the branch that I want to merge I assume you mean I should not request a merge into master but into the naze-branch instead. I changed the pull-request accordingly

I will integrate your comments soon.

**Comment by racodond** [comment]
> Did you make this rule work in real life? I mean unit testing apart.
> What do mean with this?

Did you start a SonarQube server with a snapshot of the plugin embedding your new rule? To check if it really works.

> I will work integrate your comments soon.

Cool!

**Comment by roookeee** [comment]
> Did you start a SonarQube server with a snapshot of the plugin embedding your new rule? To check if it really works.

I did not do this yet, but will do so on monday

**Comment by roookeee** [comment]
I have checked the rule in a fresh sonarqube-6.7 installation. it seems to be working
image

**Comment by roookeee** [comment]
SelectorNumberTest fails on my Windows 10 machines. I suspect a decimal-format-locale difference:

> SelectorNumberTest.test:30 [Bad message at line 0] expected:<...his sheet contains 4[,097 selectors, 2 more than the 4,]095 maximum.'> but was:<...his sheet contains 4[.097 selectors, 2 more than the 4.]095 maximum.'>

Notice the '.' instead of ','

**Comment by roookeee** [comment]
I fixed the failing integration tests besides SelectorNumberTest.
Can you help me with CssTest.java? I do not really understands whats going on there and why it is failing

**Comment by racodond** [comment]
@naze: Thanks for your work! I'll merge the PR into the naze branch and I'll fix the integration tests. I'll make you review it before merging to master.

**Pull request merged**